argocd/iac/terraform/examples/eks/private-git/main.tf

```diff
@@ -0,0 +1,299 @@
+    #enable_aws_for_fluentbit                     = true
+    #enable_aws_node_termination_handler          = true
+    #enable_karpenter                             = true
+    #enable_velero                                = true
+    #enable_aws_gateway_api_controller            = true
+    #enable_aws_ebs_csi_resources                 = true # generate gp2 and gp3 storage classes for ebs-csi
+    #enable_aws_secrets_store_csi_driver_provider = true
+  }
+  oss_addons = {
+    #enable_argo_rollouts                         = true
+    #enable_argo_workflows                        = true
+    #enable_cluster_proportional_autoscaler       = true
+    #enable_gatekeeper                            = true
+    #enable_gpu_operator                          = true
+    #enable_ingress_nginx                         = true
+    #enable_kyverno                               = true
+    #enable_kube_prometheus_stack                 = true
+    enable_metrics_server = true
+    #enable_prometheus_adapter                    = true
+    #enable_secrets_store_csi_driver              = true
+    #enable_vpa                                   = true
+    #enable_foo                                   = true # you can add any addon here, make sure to update the gitops repo with the corresponding application set
+  }
+  addons = merge(local.aws_addons, local.oss_addons, { kubernetes_version = local.cluster_version })
+
+  addons_metadata = merge(
+    module.eks_blueprints_addons.gitops_metadata,
+    {
+      aws_cluster_name = module.eks.cluster_name
+      aws_region       = local.region
+      aws_account_id   = data.aws_caller_identity.current.account_id
+      aws_vpc_id       = module.vpc.vpc_id
+    },
+    {
+      gitops_bridge_repo_url      = "${local.gitops_addons_org}/${local.gitops_addons_repo}"
+      gitops_bridge_repo_revision = local.gitops_addon_revision
+    }
+  )
+
+  argocd_bootstrap_app_of_apps = {
+    addons = templatefile("${path.module}/bootstrap/addons.yaml", {
+      repoURL        = "${local.gitops_addons_org}/${local.gitops_addons_repo}"
+      path           = local.gitops_addon_path
+      targetRevision = local.gitops_addon_revision
+    })
+    workloads = templatefile("${path.module}/bootstrap/workloads.yaml", {
+      repoURL        = "${local.gitops_workloads_org}/${local.gitops_workloads_repo}"
+      path           = local.gitops_workloads_path
+      targetRevision = local.gitops_workloads_revision
+    })
+  }
+
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 3)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/csantanapr/terraform-gitops-bridge"
+  }
+}
+
+################################################################################
+# GitOps Bridge: Private ssh keys for git
+################################################################################
+resource "kubernetes_namespace" "argocd" {
```

Comment:
I've got some issues with several terraform apply complaining that the namespace already exists..

Reply:
the namespace should not exist, in the bootstrap I added for the helm to not create the namespace because it already exists 

```hcl
module "gitops_bridge_bootstrap" {
  source = "../../../modules/gitops-bridge-bootstrap"

  argocd_cluster               = module.gitops_bridge_metadata.argocd
  argocd_bootstrap_app_of_apps = local.argocd_bootstrap_app_of_apps
  argocd = { create_namespace = false }
  depends_on = [kubernetes_secret.git_secrets]
}

```
the `argocd = { create_namespace = false }`

Reply:
what I'm having trouble is with the destroy now of the namespace